src/Microsoft.Framework.ConfigurationModel.Json/JsonConfigurationExtension.cs

```diff
@@ -1,13 +1,33 @@
 // Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using Microsoft.Framework.ConfigurationModel.Json;
+using System;
+using System.IO;
+
 namespace Microsoft.Framework.ConfigurationModel
 {
     public static class JsonConfigurationExtension
     {
         public static IConfigurationSourceRoot AddJsonFile(this IConfigurationSourceRoot configuration, string path)
         {
-            configuration.Add(new JsonConfigurationSource(path));
+            return AddJsonFile(configuration, path, false);
+        }
+
+        public static IConfigurationSourceRoot AddJsonFile(this IConfigurationSourceRoot configuration, string path, bool optional)
+        {
+            if (string.IsNullOrEmpty(path))
+            {
+                throw new ArgumentException(Resources.Error_InvalidFilePath, "path");
+            }
+
+            var fullPath = PathResolver.ResolveAppRelativePath(path);
+
+            if (!optional || File.Exists(fullPath))
```

Comment:
We probably should  throw here if the file doesn't exist. We let it flow to the call to load the file at the moment, which doesn't really cause any issues but the line reported on the diagnostics page would be better if we throw here I think. Since we are now doing an exists check every time anyway we might as well catch it here and show people an error. @davidfowl do you agree?